torch/_dynamo/variables/torch.py

```diff
@@ -1565,6 +1565,23 @@ def call_function(
             if isinstance(out_kwarg_vt, variables.TensorVariable):
                 saved_out_shapes = out_kwarg_vt.proxy.node.meta["example_value"].shape
 
+        from torch.utils._device import _device_constructors
+
+        # Check if we are calling a torch factory function which
+        # accepts device as a kwarg
+        if (
```

Comment:
Why isn't line 1440 getting hit above? we should be dispatching through there and not have this custom logic. The torch function mode should be handling this kwarg.

I'm thinking that the device_constructors aren't in the list of overridable functions when we can check if torch function should be dispatched. I'd say either add these functions to that list, or  additionally check if the function is in your device_constructors list.